realm/realm-library/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -65,6 +68,24 @@ public void onResult(int count) {
         }
     }
 
+    public static class WatchDogHandler extends Handler {
+        public WatchDogHandler(Looper looper, final TestCase testCase) {
+            this(looper, testCase, TIMEOUT_IN_MILLISECONDS);
+        }
+
+        public WatchDogHandler(Looper looper, final TestCase testCase, long timeout) {
+            super(looper);
+            postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    // Timeout
+                    testCase.fail("Timeout happens.");
```

Comment:
since you're using the same Looper (`InstrumentationThread` looper) if the _test1_ complete_ correctly, you still have this Runnable in the queue, which will be processed once you resume using the looper for the _test2_